mintlify/mint
# Hahnbee/min 1386 make reusable components work for multi-tenancy

## Summary
Snippets are reusable components/chunks of text. In order to create a snippet, it must live in the `_snippets` folder. In order to reference a snippet in content you must follow the following syntax: `<Snippet file="path/filename.mdx" />`. There is no need to import the component/file. More details about the product/tech spec [here](https://docs.google.com/document/d/1bb0OsoJoWAplFvwgmdoe1yuL7RUXPOumDRbkAMHfhgg/edit?usp=sharing)

- Takes in a snippets array in getStaticProps
- If a  Snippet component (`<Snippet file="path/filename.mdx" />`) is found in the content, it replaces it with the proper snippet.
- `remarkMdxInjectSnippet.ts` is probably the golden standard for plugins now. It's the best written plugin thanks to [wooorm](https://github.com/wooorm) ref: https://github.com/mdx-js/mdx/discussions/2210

### To-Do's
- [ ] Debug hydration error from react email's snippet
- [ ] Create PR's for react email and elementary and migrate them to VP
- [ ] Reflect this set-up in single tenancy
- [ ] Write docs for snippet
- [ ] Update CLI

## Test Plan
1. Check out to the corresponding backend PR branch `hahnbee/min-1386-make-reusable-components-work-from`
2. Navigate to snippet-test.localhost:3000 and make sure it looks the same with https://react.email/docs/introduction You can see the pages that have snippets by looking at [the changes I made in the contents] (https://github.com/zenorocha/react-email/compare/main...hahnbeelee:react-email:main).
3. Navigate to elementary.localhost:3000 and make sure it looks the same with https://docs.elementary-data.com/introduction You can see the pages that have snippets by looking at [the changes I made in the contents](https://github.com/elementary-data/elementary/compare/master...hahnbeelee:elementary:master)
